RIOT-OS/RIOT
# sys/posix_sleep: Add Kconfig support

### Contribution description

This models the `posix_sleep` module in Kconfig.

### Testing procedure

```
make clean -C tests/posix_sleep/ 
TEST_KCONFIG=1 make menuconfig -C tests/posix_sleep/ 
TEST_KCONFIG=1 make all test -C tests/posix_sleep/
```

And check the CI

### Issues/PRs references
